modular_skyrat/modules/reagent_forging/code/crafting_bench.dm

```diff
@@ -11,82 +9,85 @@
-	var/static/list/allowed_choices = list(
-		"Plate Helmet" = /obj/item/clothing/head/helmet/forging_plate_helmet,
-		"Plate Armor" = /obj/item/clothing/suit/armor/forging_plate_armor,
-		"Plate Gloves" = /obj/item/clothing/gloves/forging_plate_gloves,
-		"Plate Boots" = /obj/item/clothing/shoes/forging_plate_boots,
-		"Horseshoes" = /obj/item/clothing/shoes/horseshoe,
-		"Ring" = /obj/item/clothing/gloves/ring/reagent_clothing,
-		"Collar" = /obj/item/clothing/neck/collar/reagent_clothing,
-		"Handcuffs" = /obj/item/restraints/handcuffs/reagent_clothing,
-		"Borer Cage" = /obj/item/cortical_cage,
-		"Pavise Shield" = /obj/item/shield/riot/buckler/reagent_weapon/pavise,
-		"Buckler Shield" = /obj/item/shield/riot/buckler/reagent_weapon,
-		"Coil" = /obj/item/forging/coil,
-		"Seed Mesh" = /obj/item/seed_mesh,
-		"Primitive Centrifuge" = /obj/item/reagent_containers/cup/primitive_centrifuge,
-		"Bokken" = /obj/item/forging/reagent_weapon/bokken,
-		"Bow" = /obj/item/forging/incomplete_bow,
+	/// What recipes are we allowed to choose from?
+	var/list/allowed_choices = list(
+		/datum/crafting_bench_recipe/plate_helmet,
+		/datum/crafting_bench_recipe/plate_vest,
+		/datum/crafting_bench_recipe/plate_gloves,
+		/datum/crafting_bench_recipe/plate_boots,
+		/datum/crafting_bench_recipe/horse_shoes,
+		/datum/crafting_bench_recipe/ring,
+		/datum/crafting_bench_recipe/collar,
+		/datum/crafting_bench_recipe/handcuffs,
+		/datum/crafting_bench_recipe/borer_cage,
+		/datum/crafting_bench_recipe/pavise,
+		/datum/crafting_bench_recipe/buckler,
+		/datum/crafting_bench_recipe/coil,
+		/datum/crafting_bench_recipe/seed_mesh,
+		/datum/crafting_bench_recipe/centrifuge,
+		/datum/crafting_bench_recipe/bokken,
+		/datum/crafting_bench_recipe/bow,
 	)
+	/// Radial options for recipes in the allowed_choices list, populated by populate_radial_choice_list
+	var/list/radial_choice_list = list()
+	/// An associative list of names --> recipe path that the radial recipe picker will choose from later
+	var/list/recipe_names_to_path = list()
+
+/obj/structure/reagent_crafting_bench/Initialize(mapload)
+	. = ..()
+	populate_radial_choice_list()
+
+/obj/structure/reagent_crafting_bench/proc/populate_radial_choice_list()
+	if(!length(allowed_choices))
+		return
+
+	if(length(radial_choice_list) && length(recipe_names_to_path)) // We already have both of these and don't need it, if this is called after these are generated for some reason
+		return
+
+	for(var/recipe as anything in allowed_choices)
+		var/datum/crafting_bench_recipe/recipe_to_take_from = new recipe()
+		var/obj/recipe_resulting_item = recipe_to_take_from.resulting_item
+		radial_choice_list[recipe_to_take_from.recipe_name] = image(icon = initial(recipe_resulting_item.icon), icon_state = initial(recipe_resulting_item.icon_state))
+		recipe_names_to_path[recipe_to_take_from.recipe_name] = recipe
+		QDEL_NULL(recipe_to_take_from)
 
 /obj/structure/reagent_crafting_bench/examine(mob/user)
```

Comment:
add a `return .` at the bottom

Reply:
I don't see the use in that?

Reply:
Oh you mean for the examine proc don't you, I don't see literally any other examine procs do that so idk about that one.